include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -0,0 +1,168 @@
+#include <range/v3/view/drop_exactly.hpp>
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/alignment/configuration/all.hpp>
+#include <seqan3/alignment/pairwise/policy/affine_gap_init_policy.hpp>
+#include <seqan3/alignment/pairwise/policy/affine_gap_policy.hpp>
+#include <seqan3/alignment/pairwise/policy/unbanded_dp_matrix_policy.hpp>
+#include <seqan3/alignment/pairwise/select_result_type.hpp>
+
+#include <seqan3/alignment/scoring/gap_scheme.hpp>
+#include <seqan3/alignment/scoring/scoring_scheme_base.hpp>
+
+#include <seqan3/core/metafunction/deferred_crtp_base.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+
+/*!\brief A function object that computes pairwise alignments given two sequences.
+ * \ingroup pairwise_alignment
+ * \tparam config_t             The configuration type; must be of type seqan3::configuration.
+ * \tparam algorithm_policies_t Template parameter pack with the policies to determine the execution of the algorithm;
+ *                              must be wrapped as seqan3::detail::deferred_crtp_base.
+ *
+ * \details
+ *
+ * Computes the actual pairwise alignment given two sequences. The correct algorithm is configured while parsing
+ * the seqan3::configuration within the detail::seqan3::configurator.
+ */
+template <typename config_t, typename ...algorithm_policies_t>
+class alignment_algorithm :
+    protected invoke_deferred_crtp_base<algorithm_policies_t, alignment_algorithm<algorithm_policies_t...>>...
+{
+public:
+    /*!\name Constructor, destructor and assignment
+     * \brief Defaulted all standard constructor.
+     * \{
+     */
+    constexpr alignment_algorithm()                                        = default;
+    constexpr alignment_algorithm(alignment_algorithm const &)             = default;
+    constexpr alignment_algorithm(alignment_algorithm &&)                  = default;
+    constexpr alignment_algorithm & operator=(alignment_algorithm const &) = default;
+    constexpr alignment_algorithm & operator=(alignment_algorithm &&)      = default;
+    ~alignment_algorithm()                                                 = default;
+
+    /*!\brief Constructs the algorithm with the passed configuration.
+     * \param cfg The configuration to be passed to the algorithm.
+     *
+     * \details
+     *
+     * The configuration is copied once to the heap during construction and maintained by a std::shared_ptr.
+     * The configuration is not passed to the function-call-operator of this function object, in order to avoid
+     * incompatible configurations between the passed configuration and the one used during configuration of this
+     * class. Further will the function object be stored in a std::function which requires copyable objects and
+     * in parallel executions the function object must be copied as well.
```

Comment:
I don't really understand this. Why do you pass a config to this object at all, and not just to `operator()`?